examples/webgl_mesh_batch.html

```diff
@@ -285,20 +294,18 @@
 		function sortFunction( list, camera ) {
 
 			// initialize options
-			this._options = this._options || {
+			const options = {
 				get: el => el.z,
-				aux: new Array( list.length ),
+				aux: aux,
 			};
 
-			const options = this._options;
 			options.reversed = this.material.transparent;
 
-			// convert depth to unsigned 32 bit range
-			const den = camera.far;
+			// bit-cast depth to unsigned 32-bit
 			for ( let i = 0, l = list.length; i < l; i ++ ) {
 
-				const el = list[ i ];
-				el.z = ( 1 << 30 ) * ( el.z / den );
+				f32[ 0 ] = list[ i ].z;
+				list[ i ].z = u32[ 0 ];
```

Comment:
> Simply normalizing the float depth is not enough to correctly sort, as the fractional part gets truncated - producing incorrect ties.

Can you elaborate on how the previous implementation is incorrect? `el.z / camera.far` converts the value to the normalized range [0, 1] and then `( 1 << 30 ) *` expands that to the full range of positive values represented by signed integers. Yes there are decimal values that will be truncated here when performing bitwise operations & int conversion but the values are truncated in this new approach when performing 32 bit float to a 64 bit float conversion, as well.

I agree the bit-casting approach is more elegant but again I think it's nice to have a single, easy-to-copy function for this case.

Reply:
While you are correct that both methods can be used correctly with sufficient care, normalization is more error prone without proper understanding. Also, the range specified `1 << 30` isn't the full range of neither signed nor unsigned int32.

```
1 << 30 // 1_073_741_824 - ~1/4th UINT32_MAX
-1 >>> 1 // 2_147_483_647 - INT32_MAX
-1 >>> 0 // 4_294_967_295 - UINT32_MAX

```

I don't mind reverting back to normalization, with the proper limits, I just thought bit-cast was even simpler.

Reply:
Sorry you're right - my brain wasn't fully working before, I think 😅 It's not even the max positive value of a signed int since only one bit is flipped. For this case why don't we use `2**32 - 1` which will be the max 32 bit uint range (unless I've missed something again 😁)